packages/task/src/browser/task-service.ts

```diff
@@ -94,33 +94,21 @@ export class TaskService implements TaskConfigurationClient {
 
         // notify user that task has finished
         this.taskWatcher.onTaskExit((event: TaskExitedEvent) => {
-            const signal = event.signal;
             if (!this.isEventForThisClient(event.ctx)) {
                 return;
             }
 
-            if (event.code === 0) {  // normal process exit
-                let success = '';
-
-                // this finer breakdown will not work on Windows.
-                if (signal && signal !== '0') {
-                    if (signal === '1') {
-                        success = 'Terminal Hangup received - ';
-                    } else if (signal === '2') {
-                        success = 'User Interrupt received - ';
-                    } else if (signal === '15' || signal === 'SIGTERM') {
-                        success = 'Termination Interrupt received - ';
-                    } else {
-                        success = 'Interrupt received - ';
-                    }
+            if (event.code !== undefined) {
+                const message = `Task ${event.taskId} has exited with code ${event.code}.`;
+                if (event.code === 0) {
+                    this.messageService.info(message);
```

Comment:
https://github.com/theia-ide/theia/issues/3604 Do not show the exit code and the signal on successful task execution?

Reply:
For now I'd prefer to leave it there, I think it's informative.  It can always be removed later.